File: src/Omnia.ProcessManagement.Models/ProcessTypes/ProcessTypeItemSettings.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Omnia.ProcessManagement.Models.ProcessTypes
{
    public class ProcessTypeItemSettings : ProcessTypeSettings
    {
        public ProcessTypeItemSettings()
        {
            ProcessTemplateIds = new List<Guid>();
        }

        public readonly static Guid ApproverGroupId = new Guid("4c32092d-21ec-4137-a3ed-50bcbf6b0f78");
        public override ProcessTypeSettingsTypes Type
        {
            get
            {
                return ProcessTypeSettingsTypes.Item;
            }
        }
        public Guid EnterprisePropertySetId { get; set; }
        public bool AllowAppendices { get; set; }
        public bool AllowConnectToTemplate { get; set; }
        public bool ReplaceTokenOnPublishing { get; set; }
        public bool AllowRevisions { get; set; }
        public bool AllowBypassApprovalForRevisions { get; set; }
        public List<Guid> ProcessTemplateIds { get; set; }
        public Guid? DefaultProcessTemplateId { get; set; }

        /// <summary>
        /// There is special value for defining approver
        /// <see cref="ApproverGroupId"/>
        /// </summary>
        public List<Guid> FeedbackRecipientsPropertyDefinitionIds { get; set; }

        /// <summary>
        /// null means no Review Reminder setup
        /// </summary>
        public ReviewReminder ReviewReminder { get; set; }

        /// <summary>
        /// null means no Archive setup
        /// </summary>
        public Archive Archive { get; set; }

        /// <summary>
        /// null means no Retention setup
        /// </summary>

        /// <summary>
        /// null means no Approval setup
        /// </summary>
        public PublishingApprovalSettings.PublishingApprovalSettings PublishingApprovalSettings { get; set; }
        public Dictionary<Guid, PropertySetItemSettings.PropertySetItemSettings> PropertySetItemSettings { get; set; }
    }
}

File: src/Omnia.ProcessManagement.Web/client/fx/models/data/processtypes/ProcessTypeItemSettings.ts
import { GuidValue, Guid } from '@omnia/fx-models';
import { ProcessTypeSettings, Archive, ReviewReminder, PublishingApprovalSettings, PropertySetItemSettings, ProcessTypeSettingsTypes } from '.';

export const ApproverId: GuidValue = "4c32092d-21ec-4137-a3ed-50bcbf6b0f78"; //remember do not new Guid here, it will cause issue in v-select, we keep it as plain text here

export interface ProcessTypeItemSettings extends ProcessTypeSettings {
    type: ProcessTypeSettingsTypes.Item;
    enterprisePropertySetId: GuidValue;
    allowRevisions: boolean;
    allowBypassApprovalForRevisions: boolean;
    processTemplateIds: Array<GuidValue>;
    defaultProcessTemplateId?: GuidValue;

    /**
     * There is a spacial value for defining approver : ApproverId - 4c32092d-21ec-4137-a3ed-50bcbf6b0f78
     * */
    feedbackRecipientsPropertyDefinitionIds: Array<GuidValue>;

    /**
     * null means no Review Reminder setup
     * */
    reviewReminder?: ReviewReminder;

    /**
     * null means no Archive setup
     * */
    archive?: Archive;

    /**
     * null means no Approval needed
     * */
    publishingApprovalSettings: PublishingApprovalSettings;

    propertySetItemSettings: { [id: string]: PropertySetItemSettings }
}